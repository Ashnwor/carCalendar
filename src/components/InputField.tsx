import React from 'react';
import { StyleSheet } from 'react-native';
import { TextInput, HelperText } from 'react-native-paper';

import FieldMandatory from './FieldMandatory';

const InputField: React.FC<{
	label: string;
	value: string | undefined;
	error: boolean;
	onChangeText: (text: string) => void;
}> = ({ label, value, error, onChangeText }) => {
	return (
		<>
			<TextInput style={styles.input} label={label} value={value} onChangeText={onChangeText} />
			{error && <FieldMandatory />}
		</>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff',
	},
	input: { margin: 5 },
	fab: {
		position: 'absolute',
		margin: 16,
		right: 0,
		bottom: 0,
	},
});

export default InputField;
