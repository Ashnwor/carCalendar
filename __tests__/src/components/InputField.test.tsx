import React from 'react';
import { render } from 'react-native-testing-library';

import InputField from '../../../src/components/InputField';

describe('<InputField />', () => {
	it('renders without crashing', () => {
		const component = render(
			<InputField
				label="test_label"
				value="test_value"
				onChangeText={(_text) => jest.fn()}
				error={false}
			/>,
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
