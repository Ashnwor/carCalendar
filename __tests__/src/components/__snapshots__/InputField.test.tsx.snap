// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {
        "margin": 5,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
      }
    }
  />
  <View
    style={
      Object {
        "minHeight": 64,
        "paddingBottom": 0,
        "paddingTop": 0,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
        }
      }
    >
      <Text
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        test_label
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        test_label
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
            "textAlign": "left",
            "zIndex": 1,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "textAlignVertical": "center",
          },
          Array [
            Object {},
          ],
        ]
      }
      underlineColorAndroid="transparent"
      value="test_value"
    />
  </View>
</View>
`;
